protocol/nats_jetstream/v3/options.go

```diff
@@ -0,0 +1,105 @@
+/*
+ Copyright 2024 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package nats_jetstream
+
+import (
+	"errors"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+)
+
+var (
```

Comment:
Tbh, I feel we're artificially increasing our API surface (coupling) with all these typed errors. If we want to expose a validation error type, why do we feel we need typed errors instead of a simplified approach using a generic validation error type with specific error messages?

Something that gives users a simple assertion approach?

```go
   var vError ValidationError
   if errors.As(err, &vError) {...}
```

Reply:
Imagine how a user would have to assert against all these errors below? Do you think there's need/someone would actually do this?

Reply:
nit: validation error type(s) should live in the protocol.go

Reply:
added validationError.  I added it as not exported type.